### ISSUE TITLE
Better response for retrieving lists

### ISSUE THREAD

**Issue by TheFern2** [TheFern2/pylogix-api#2]
Would like a better json response when doing lists, not just a list values.

Currently it returns this:
```
[
  "BaseBOOL = True",
  "BaseDINT = -545437484",
  "BaseINT = 6512"
]
```

Would like this instead:
```json
 {
   "tags": {
       "tag": {
           "status": "Success",
           "tagName": "BaseBOOL",
           "value": true
       },
       "tag": {
           "status": "Success",
           "tagName": "BaseDINT",
           "value": 456464
       }
   }
 }
```

**Comment by GitHubDragonFly** [comment]
Just as a quick fix for the current code, you can add string descriptions and also add a Status to the output but it will all have to be inlined:

            for i in range(0, len(ret)):
                result.append('TagName: ' + str(ret[i].TagName) + ' Value: ' + str(ret[i].Value) + ' Status:  ' + str(ret[i].Status))

A possible stacked up way would be with multiple appends, like this (not tested by me yet):

            for i in range(0, len(ret)):
                result.append('TagName: ' + str(ret[i].TagName))
                result.append( Value: ' + str(ret[i].Value))
                result.append(' Status:  ' + str(ret[i].Status))
                result.append('---------------------------')

**Comment by GitHubDragonFly** [comment]
You might also consider updating your description for reading arrays since what works for 1-dimensional array also works for 2D and 3D arrays:

- localhost:5000/pylogix/v1.0/plc/192.168.0.10/0/tags/someArrayTag[x]{a}
- localhost:5000/pylogix/v1.0/plc/192.168.0.10/0/tags/someArrayTag[x, y]{a}
- localhost:5000/pylogix/v1.0/plc/192.168.0.10/0/tags/someArrayTag[x,y,z]{a}

**Comment by GitHubDragonFly** [comment]
Here is a version that actually does it the way you wanted (I just used the code from your get_all_tags).

You should carefully review it since I also added the links to other pages to each of the functions (which you can remove if you wish).

[app.zip](https://github.com/TheFern2/pylogix-api/files/6126582/app.zip)

**Comment by TheFern2** [comment]
@GitHubDragonFly thanks will look at it when I get a chance. 😃

**Comment by GitHubDragonFly** [comment]
Here is another version with the following features:

- Out of the box, it works as you originally intended.
- A new addition is to allow displaying any of the pages in a table. It requires adding the "templates" folder and the "TableList.html" file to the project.
- A boolean flag was added to control showing a table in each of the functions. Probably the best use of the table would be for tag listing.
- A boolean flag was added to control showing links in each of the functions, which has no effect if the table option is selected.

You should carefully review it and decide if any of it is worth including in the repo.

[pylogix-api.zip](https://github.com/TheFern2/pylogix-api/files/6135871/pylogix-api.zip)

**Comment by GitHubDragonFly** [comment]
Here is another one but with the BUG fix (for the same mistake I was making previously) and a new feature:

- BUG fix, use semicolon instead of comma for multiple tag separation so multi-dimensional arrays could be included:
  - localhost:5000/pylogix/v1.0/plc/192.168.1.15/3/tags/[CT_DINT; CT_REAL; CT_STRING; CT_DINT_3D[0,0,0]]

- New Feature, allow reading elements of a single or multi dimensional array when included in array of tags (this is using multiple consecutive reads)
  - localhost:5000/pylogix/v1.0/plc/192.168.1.15/3/tags/[CT_DINT; CT_REAL; CT_STRING; CT_BOOLArray[0]{25}; CT_DINT_3D[0,1,0]{3}]

[pylogix-api.zip](https://github.com/TheFern2/pylogix-api/files/6135959/pylogix-api.zip)

**Comment by GitHubDragonFly** [comment]
Here is the TableList.html template file which shows an example of how to add styles to the table.

[TableList.zip](https://github.com/TheFern2/pylogix-api/files/6136047/TableList.zip)

The resulting table pictures are attached.

![Tag Table (BOOL True False)](https://user-images.githubusercontent.com/69928501/111102151-c6c2c400-8521-11eb-9f2e-fb3055e1af4d.png)

![Tag Table (BOOL One Zero)](https://user-images.githubusercontent.com/69928501/111102152-c75b5a80-8521-11eb-9eb8-39604d734ae7.png)

**Comment by GitHubDragonFly** [comment]
Just to mention that what applies to reading elements of an array also works for reading bits, for example: CT_DINT.0{15}.

**Comment by GitHubDragonFly** [comment]
Here is the latest app but with a new feature which allows showing Bit/Bool values as either True : False or 1 : 0.

[app.zip](https://github.com/TheFern2/pylogix-api/files/6139068/app.zip)

A new bool variable "showBoolAsOneZero" was added to control this new feature.

The pictures of the tables in my previous post were updated to reflect these latest changes.

This is all for when you find time to review them.

**Comment by GitHubDragonFly** [comment]
I just reduced some code and reset all boolean control values to False so the app starts the way you wanted it initially.
The file in my previous post was replaced.

**Comment by GitHubDragonFly** [comment]
Some more changes, added missing "showBoolAsOneZero" code for single tag reading and some error checking in all functions for when not connected. The file in my previous post was replaced.

**Comment by TheFern2** [comment]
Hey @GitHubDragonFly thank you very much. There are quite a few changes, please submit PR when you get a chance no rush, if you're having git issues let me know I can help in the workflow if you'd like. I do think is a good idea to have HTML in an examples folder, but I want the api to be focused strictly on the backend so whoever uses this can use whatever front end they choose to.

**Comment by TheFern2** [comment]
> Hey @GitHubDragonFly thank you very much. There are quite a few changes, please submit PR when you get a chance no rush, if you're having git issues let me know I can help in the workflow if you'd like. I do think is a good idea to have HTML in an examples folder, but I want the api to be focused strictly on the backend so whoever uses this can use whatever front end they choose to.

Ah never mind, I just saw the boolean flag. : )

**Comment by TheFern2** [comment]
fixed by #3
